src/libraries/System.Diagnostics.PerformanceCounter/src/System/Diagnostics/SharedPerformanceCounter.cs

```diff
@@ -697,6 +697,7 @@ private unsafe CategoryData GetCategoryData()
                             }
                             else
                             {
+                                Debug.Assert(counterNamesObject is string[], $"Expected string[], got '{counterNamesObject}' for {_categoryName}");
                                 string[] counterNames = (string[])counterNamesObject;
```

Comment:
do you need to log `categoryKey.GetValueKind() ` or does it include it in ToString() ?